Sources/Private/Utility/Interpolatable/KeyframeGroup+Extensions.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  KeyframeGroup+Extensions.swift
+//  Lottie
+//
+//  Created by JT Bergman on 6/20/22.
+//
+
+import Foundation
+
+extension KeyframeGroup where T == Vector1D {
+  /// Manually interpolates the keyframes so that they are defined linearly
+  ///
+  /// This method uses `UnitBezier` to perform the interpolation. It will create one keyframe
+  /// for each frame of the animation. For instance, if it is given a keyframe at time 0 and a keyframe
+  /// at time 10, it will create 10 interpolated keyframes. It is currently not optimized.
```

Comment:
> It is currently not optimized

The gradient strokes animation seems to use 5% CPU with Core Animation vs 15% CPU with main thread. The Duolingo animations CPU usage isn't changed much since there is now one keyframe. There's room for optimization here, but this PR is already quite long. The simplest idea is that a new keyframe doesn't need to created if the following is true: 
```
// Linear change can be collapsed into two keyframes instead of 3
curr.newValue - prev.newValue == prev.newValue - prev.prev.newValue
```

Considering the CPU usage, it may be worth still throwing an error to warn the user?

Reply:
This just happens once when setting up the animation, right? As long as we don't incur a performance hit on every frame (like with the main thread engine) then this seems perfectly fine to me!

Reply:
That test is a bit flaky, I'd just suggest increasing the threshold from `0.5` to `0.55`